spec/OneClickInstall_spec.rb

```diff
@@ -52,13 +62,100 @@
   describe "#SetRequiredRepository" do
     context "set repository to recommended" do
       it "Ensures that the repository with the specified URL is selected for addition" do
-        url = "http://download.opensuse.org/tumbleweed/repo/oss/"
-        subject.instance_variable_set(:@repositories, { url => { "name" => "vim" , "recommended" => "false" }})
-        subject.SetRequiredRepository(url)
+        subject.instance_variable_set(:@repositories, { urls[0] => { "name" => "oss" , "recommended" => "false" }})
+        subject.SetRequiredRepository(urls[0])
         repos = subject.instance_variable_get(:@repositories)
-        expect(repos[url]["name"]).to eql("vim")
-        expect(repos[url]["recommended"]).to eql("true")
+        expect(repos[urls[0]]["name"]).to eql("oss")
+        expect(repos[urls[0]]["recommended"]).to eql("true")
       end
     end
   end
+
+    describe "#SetNonRequiredRepository" do
+    context "set repository to not recommended" do
+      it "Ensures that the repository with the specified URL is not selected for addition" do
+        subject.instance_variable_set(:@repositories, { urls[1] => { "name" => "non-oss" , "recommended" => "true" }})
+        subject.SetNonRequiredRepository(urls[1])
+        repos = subject.instance_variable_get(:@repositories)
+        expect(repos[urls[1]]["name"]).to eql("non-oss")
+        expect(repos[urls[1]]["recommended"]).to eql("false")
+      end
+    end
+  end
+
+  describe "#SetRequiredRepositories" do
+    context "set repositores to recommended" do
+      it "Ensures that the repositories with the specified URL are selected for addition and all others are not" do
+        subject.SetRequiredRepositories(urls)
+        repos = subject.instance_variable_get(:@repositories)
+        expect(repos[urls[0]]["name"]).to eql("oss")
+        expect(repos[urls[0]]["recommended"]).to eql("true")
```

Comment:
is already true before running `SetRequiredRepositories` I guess we should switched to know for sure that something changes.

Reply:
don't we need a 3rd one here to probe that others are untouched?

Reply:
Non-oss repo is initially set to false, while oss repository is initially set to true. So the function "SetRequiredRepositories" is expected to change only non-oss, which indeed is what happens

Reply:
I'm referring to the example description '...and all others are not', so if you want to test that, you need at least 3, so if you pass those two, the other one stays unchanged.

Reply:
You mean that we should make sure that a repo that is not even given as an argument to the function, should not be changed? I find this a very extreme. I prefer not to add such a check.

Reply:
yes, I agree, we are not testing negative cases neither, I just mentioned to you because of the description seem to mean that...maybe a better description for the example `it` ?

Reply:
I agree with you. I copied that description from the actual code description, but indeed is not very clear. I can change it in the unit test